Error trying to build datatool

## 🐛 Bug

<!-- A clear and concise description of what the bug is. -->
I am trying to get the semantic information for the Gibson dataset. One of these steps is to run `tools/gen_gibson_semantics.sh` which has a line that runs `build/utils/datatool/datatool`. However, I noticed I didn't have this file in my build, so I reran my build and noticed I was getting an error while this was trying to be built "/path/to/habitat-sim/src/utils/datatool/SceneLoader.cpp:34:22: error: ‘class esp::assets::GenericInstanceMeshData’ has no member named ‘loadPLY’". It looks like the class GenericInstanceMeshData used to have loadPLY, but now it does not. 

I have gotten around this issue by using release 0.14, building that, and using the generated build to create my desired files for semantic information on the Gibson dataset. I also had to move over `./tools/gen_gibson_semantics.sh`.
## Command
`python setup.py build --build-datatool`

## To Reproduce

Steps to reproduce the behavior:

<!-- If you were running a command, post the exact command that you were running -->

1. git clone https://github.com/facebookresearch/habitat-sim.git 
2. cd habitat-sim 
3. conda activate topological (This is my anaconda env with all of the requirements already installed, it also has a build of habitat-sim/habitat-api installed onto it)
4. python setup.py build --build-datatool

<!-- If you have a code sample, error messages, stack traces, please provide it here as well -->

## Expected behavior

I expected 'build/utils/datatool/datatool' to be generated.

## Additional context

<!-- Add any other context about the problem here. -->

## System Info

Please copy and paste the output from the
[environment collection script](https://github.com/facebookresearch/habitat-sim/tree/master/habitat_sim/utils/collect_env.py)
(or fill out the checklist below manually).

You can run the script with:
```
# For security purposes, please check the contents of collect_env.py before running it.
python habitat_sim/utils/collect_env.py
```

ENVIRONMENT INFO:
Platform: Linux-5.3.0-51-generic-x86_64-with-debian-buster-sid
Machine: x86_64
Processor: x86_64
Libc version: glibc 2.9
Mac version: 
Python version: 3.6.10
Architecture: 64bit 
Win version:    
System OS: Linux
Release: 5.3.0-51-generic
Version: #44~18.04.2-Ubuntu SMP Thu Apr 23 14:27:18 UTC 2020
Operational System: linux
GCC version: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
CMAKE version: cmake version 3.14.0
NVIDIA-SMI: Thu Jun 11 15:12:41 2020       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.59       Driver Version: 440.59       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 2080    Off  | 00000000:07:00.0 Off |                  N/A |
|  0%   38C    P8    10W / 225W |      1MiB /  7982MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce RTX 2080    Off  | 00000000:41:00.0 Off |                  N/A |
|  0%   43C    P8    17W / 225W |    557MiB /  7979MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
Pip packages versions:
matplotlib==2.1.1
numpy==1.18.0
tqdm==4.42.1
Conda packages versions:
attrs                     19.3.0                   pypi_0    pypi  19.1.0
gitpython                 3.1.2                    pypi_0    pypi
matplotlib                3.2.1                    pypi_0    pypi
numba                     0.49.1                   pypi_0    pypi
numpy                     1.18.4                   pypi_0    pypi
numpy-quaternion          2020.5.19.15.27.24          pypi_0    pypi
pillow                    7.1.2            py36hb39fc2d_0  
scipy                     1.4.1                    pypi_0    pypi  1.3.0
tqdm                      4.46.0                   pypi_0    pypi

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -342,7 +342,12 @@ jobs:
 
               export PYTHONPATH=$PYTHONPATH:$(pwd)
               # run tests with code coverage
-              CORRADE_TEST_COLOR=ON GTEST_COLOR=yes ./build.sh --headless --bullet --with-cuda --run-tests --cmake-args='-DCMAKE_CXX_FLAGS="--coverage"'
+              CORRADE_TEST_COLOR=ON GTEST_COLOR=yes ./build.sh --headless \
+                --bullet \
+                --with-cuda \
+                --build--datatool \
+                --run-tests \
+                --cmake-args='-DCMAKE_CXX_FLAGS="--coverage"'
 
               while [ ! -f ~/miniconda/pytorch_installed ]; do sleep 2; done # wait for Pytorch
               pytest --cov-report=xml --cov=./
@@ -424,7 +429,7 @@ jobs:
               export PATH=$HOME/miniconda/bin:/usr/local/cuda/bin:$PATH
               . activate habitat; cd habitat-api
               python setup.py develop --all
-              
+
               cd docs
               conda install -y -c conda-forge doxygen==1.8.16
               conda install -y  jinja2 pygments docutils
@@ -459,7 +464,7 @@ jobs:
                   rm -rf published/docs/${dir}
                   cp -r ../habitat-sim/build/docs-public/${dir} published/docs/.
               done
-              
+
               git config --global user.name "Habitat"
               git config --global user.email habitat@fb.com
               NOW=$(date +"%m-%d-%Y")

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -345,7 +345,7 @@ jobs:
               CORRADE_TEST_COLOR=ON GTEST_COLOR=yes ./build.sh --headless \
                 --bullet \
                 --with-cuda \
-                --build--datatool \
+                --build-datatool \
                 --run-tests \
                 --cmake-args='-DCMAKE_CXX_FLAGS="--coverage"'
 

--- a/src/utils/datatool/SceneLoader.cpp
+++ b/src/utils/datatool/SceneLoader.cpp
@@ -19,8 +19,20 @@
 #include <assimp/scene.h>
 #include <assimp/Importer.hpp>
 
+#include <Magnum/Trade/AbstractImporter.h>
+
+namespace Cr = Corrade;
+
 namespace esp {
 namespace assets {
+
+SceneLoader::SceneLoader()
+    :
+#ifdef MAGNUM_BUILD_STATIC
+      // avoid using plugins that might depend on different library versions
+      importerManager_("nonexistent")
+#endif
+          {};
 
 MeshData SceneLoader::load(const AssetInfo& info) {
   MeshData mesh;
@@ -30,12 +42,15 @@ MeshData SceneLoader::load(const AssetInfo& info) {
   }
 
   if (info.type == AssetType::INSTANCE_MESH) {
-    GenericInstanceMeshData instanceMeshData;
-    instanceMeshData.loadPLY(info.filepath);
+    Cr::Containers::Pointer<Importer> importer;
+    CORRADE_INTERNAL_ASSERT_OUTPUT(
+        importer = importerManager_.loadAndInstantiate("StanfordImporter"));
+    GenericInstanceMeshData::ptr instanceMeshData =
+        GenericInstanceMeshData::fromPLY(*importer, info.filepath);
 
-    const auto& vbo = instanceMeshData.getVertexBufferObjectCPU();
-    const auto& cbo = instanceMeshData.getColorBufferObjectCPU();
-    const auto& ibo = instanceMeshData.getIndexBufferObjectCPU();
+    const auto& vbo = instanceMeshData->getVertexBufferObjectCPU();
+    const auto& cbo = instanceMeshData->getColorBufferObjectCPU();
+    const auto& ibo = instanceMeshData->getIndexBufferObjectCPU();
     mesh.vbo = vbo;
     mesh.ibo = ibo;
     for (const auto& c : cbo) {

--- a/src/utils/datatool/SceneLoader.h
+++ b/src/utils/datatool/SceneLoader.h
@@ -6,16 +6,30 @@
 
 #include <string>
 
+#include <Corrade/PluginManager/Manager.h>
+
 #include "esp/assets/Asset.h"
 #include "esp/assets/CollisionMeshData.h"
 #include "esp/assets/MeshData.h"
+
+namespace Magnum {
+namespace Trade {
+class AbstractImporter;
+}
+}  // namespace Magnum
 
 namespace esp {
 namespace assets {
 
 class SceneLoader {
+  using Importer = Magnum::Trade::AbstractImporter;
+
  public:
+  SceneLoader();
   MeshData load(const AssetInfo& info);
+
+ private:
+  Corrade::PluginManager::Manager<Importer> importerManager_;
 };
 
 }  // namespace assets

